test/unit/logged_data/test_queuing_logged_data_destination.py

```diff
@@ -0,0 +1,52 @@
+from typing import Any, Optional
```

Comment:
It's likely hard to unit-test, but looks like there are a few untested behaviours (flushing, timeout, etc)